cmd/bosun/web/web.go

```diff
@@ -90,56 +103,80 @@ func Listen(listenAddr string, devMode bool, tsdbHost string, reloadFunc func()
 		}
 	}
 
+	baseChain := alice.New(miniprofilerMiddleware, endpointStatsMiddleware, gziphandler.GzipHandler)
+
+	auth, tokens, err := buildAuth(authConfig)
+	if err != nil {
+		slog.Fatal(err)
+	}
+
+	//helpers to add routes with middleware
+	handle := func(route string, h http.Handler, perms easyauth.Role) *mux.Route {
+		return router.Handle(route, baseChain.Then(auth.Wrap(h, perms)))
+	}
+	handleFunc := func(route string, h http.HandlerFunc, perms easyauth.Role) *mux.Route {
+		return handle(route, h, perms)
+	}
+
+	const (
+		GET  = http.MethodGet
+		POST = http.MethodPost
+	)
+
 	if tsdbHost != "" {
-		router.HandleFunc("/api/index", IndexTSDB)
-		router.Handle("/api/put", Relay(tsdbHost))
+		handleFunc("/api/index", IndexTSDB, canPutData).Name("tsdb_index")
+		handle("/api/put", Relay(tsdbHost), canPutData).Name("tsdb_put")
 	}
+	router.PathPrefix("/auth/").Handler(auth.LoginHandler())
+	handleFunc("/api/", APIRedirect, fullyOpen).Name("api_redir")
+	handle("/api/action", JSON(Action), canPerformActions).Name("action").Methods(POST)
+	handle("/api/alerts", JSON(Alerts), canViewDash).Name("alerts").Methods(GET)
+	handle("/api/config", JSON(Config), canViewConfig).Name("get_config").Methods(GET)
 
-	router.HandleFunc("/api/", APIRedirect)
-	router.Handle("/api/action", JSON(Action))
-	router.Handle("/api/alerts", JSON(Alerts))
-	router.Handle("/api/config", miniprofiler.NewHandler(Config))
-	router.Handle("/api/config_test", miniprofiler.NewHandler(ConfigTest))
-	router.Handle("/api/save_enabled", JSON(SaveEnabled))
-	if schedule.SystemConf.ReloadEnabled() { // Is true of save is enabled
-		router.Handle("/api/reload", JSON(Reload)).Methods(http.MethodPost)
+	handle("/api/config_test", JSON(ConfigTest), canViewConfig).Name("config_test").Methods(POST)
+	handle("/api/save_enabled", JSON(SaveEnabled), fullyOpen).Name("seve_enabled").Methods(GET)
+
+	if schedule.SystemConf.ReloadEnabled() {
+		handle("/api/reload", JSON(Reload), canSaveConfig).Name("can_save").Methods(POST)
 	}
+
 	if schedule.SystemConf.SaveEnabled() {
```

Comment:
I like that we are more specific about the methods. But may need to note is as a break change if people are doing odd things.